src/main/java/frc/robot/utils/AutoAlignment.java

```diff
@@ -4,44 +4,90 @@
 import edu.wpi.first.math.controller.PIDController;
 import edu.wpi.first.math.geometry.Pose2d;
 import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
 import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import frc.robot.RobotContainer;
 import frc.robot.constants.Constants;
+import frc.robot.subsystems.Ramp;
+import frc.robot.utils.math.VectorUtils;
+import frc.robot.utils.math.VelocityVector;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.function.BiFunction;
 
 public class AutoAlignment {
     private final static HashMap<Alignable, BiFunction<Double, Double, Rotation2d>> positionAngleMap = new HashMap<>(Map.of(
-       Alignable.AMP, (x,y)-> new Rotation2d(Math.PI/2),
-       Alignable.SPEAKER,(x,y)-> new Rotation2d(RobotContainer.isRedAlliance() ? 0 : Math.PI).plus(new Rotation2d(Math.atan(y/x))
-    )));
-    public static double calcTurnSpeed(Alignable alignable, Rotation2d currentAngle, double x, double y, PIDController controller){
-        double targetAngle = getAngle(alignable,x,y).getDegrees();
-        if (Math.abs(currentAngle.getDegrees() - targetAngle) < Constants.AUTO_ALIGN_THRESHOLD){
+            Alignable.AMP, (x, y) -> new Rotation2d(Math.PI / 2),
```

Comment:
This is too complicated to be able to follow. Needs to be simplified.

Reply:
Did you have an idea in mind?

Reply:
I do. Let's talk in person once I'm in the shop.